src/lib/journey/config.store.ts

```diff
@@ -0,0 +1,72 @@
+import { readable, type Readable } from 'svelte/store';
+import { z } from 'zod';
+
+export const journeyItemSchema = z.object({
+  journey: z.string(),
+  key: z.string(),
+  match: z.string().array(),
+});
+export const journeysSchema = z.object({
+  default: journeyItemSchema,
+  forgotPassword: journeyItemSchema,
+  forgotUsername: journeyItemSchema,
+  login: journeyItemSchema,
+  register: journeyItemSchema,
+});
+
+export const partialJourneysSchema = journeysSchema.partial();
+
+export let configuredJourneys: Readable<z.infer<typeof journeyItemSchema>[]>;
+
+/**
+ * Change `any` here to something that actually works
```

Comment:
I took a whack at this cause I was bored tonight. I made some small changes, but if we better unify the keys of the object to the `key` property wthin the nested object and journey we can write a neat type.

```
type UncapitalizeObjectKeys<T> = {
  [key in keyof T as Uncapitalize<key & string>]: T[key] extends Object ? UncapitalizeObjectKeys<T[key]> : T[key]
}
type Journey = 'ResetPassword' | 'ForgottenUsername' | 'Registration' | 'Login'
type JourneyKeys = keyof UncapitalizeObjectKeys<{
    [K in Journey]: K
}>

type Key = Lowercase<Journey>
type Match = `#/service/${Journey}`

type Setup = { journey?: Journey, key: JourneyKeys, match: string[] }

const defaultJourneys = {
  resetPassword: {
    journey: 'ResetPassword',
    key: 'resetPassword',
    match: ['#/service/ResetPassword'],
  },
  forgottenUsername: {
    journey: 'ForgottenUsername',
    key: 'forgottenUsername',
    match: ['#/service/ForgottenUsername'],
  },
  login: {
    journey: undefined,
    key: 'login',
    match: ['#/service/Login', '?journey'],
  },
  registration: {
    journey: 'Registration',
    key: 'registration',
    match: ['#/service/Registration', '?journey=Registration'],
  },
} satisfies  Record<JourneyKeys, Setup>;
```

https://www.typescriptlang.org/play?ssl=35&ssc=41&pln=1&pc=1#code/C4TwDgpgBAqgdgYwIZgJbCQG1QLwgeQCMArCBYAaQhAGcAeAFQD4oBeKAbwCgooBtANbUoqOFCEgA9gDMoDKEhqxEKdFlwQ6EqADIoNYACdRAcyYBdAFxzB1c1AgAPYBDgATJUVLkoAfmXIaBjYeF5klNT0DLYg5izW0RLmXAC+XKCQUABSkgCuhnDC7ADkAEoQNBDAAAqKNADukoZuxVAAPlDFAGJNJpLALnAwlQVIALYQrR1lECaoBoZIwKiScFOdADKSc2vp4NA5+YUgVLRs4tQyAarBGmHkp-TcvLx8FCJihwV21hSpTFw9plTuctvUIIZkJU6F9jgCMtAALJLBAAC3OAAMAMQAehGADdUAgIDiACQcWHUFIYwEIqAAZSquTA5w4UGIeW+IF81kpIAANBcQLzOcdHoKxijUdYFqY+PY0lwEKsDFA3BBpEhcphgHylOxnlBDBUqrUaA0mm5rIbeByjtRrDNKjU6o1msV+TwXhJHcbnWaLe7PS8oJLgGjrHxiriCUSSeV-a7LcVzMGoCk09Jev1BsMIXBxhBrV7baKHZ0eoY+gNXHnRhMPSWhY6s1Wc7WRgWG2neGGI-xo3iIYTiTjK9Xc53Cym0xmvZhtqJiyG7VzrLl3BrRBA3D3m50FztGyG+9KBzHh3GcVsj4Lir5V8cZ1657xjXMFksVnBly9H+WZg-Iwv1WY9vQA995mA5ZQL3U9I0HWNR3KIDFhgtY7wfMsQFYFCoLQ79n14OcUn0L8aGkVAKl4cplWaGFsPFBkmTAJgAG4gA

Reply:
I think if we have something like ts-toolbelt we can make this work easier / better . I think.

Reply:
I also think we may be able to improve the type of `match` but i didn't wanna go down that road.

Reply:
I did it with typing `Match` a bit, again small changes are in here but it was fun. let me know if this is possible with these changes.

```typescript
import { readable, type Readable } from 'svelte/store';
import { z } from 'zod';

export const journeyItemSchema = z.object({
  journey: z.string(),
  key: z.string(),
  match: z.string().array(),
});
export const journeysSchema = z.object({
  default: journeyItemSchema,
  forgotPassword: journeyItemSchema,
  forgotUsername: journeyItemSchema,
  login: journeyItemSchema,
  register: journeyItemSchema,
});
export const partialJourneysSchema = journeysSchema.partial();
type UncapitalizeObjectKeys<T> = {
  [key in keyof T as Uncapitalize<key & string>]: T[key] extends Record<string, unknown> ? UncapitalizeObjectKeys<T[key]> : T[key]
}
type Journey = 'ResetPassword' | 'ForgottenUsername' | 'Registration' | 'Login'
type JourneyKeys = keyof UncapitalizeObjectKeys<{
  [K in Journey]: K
}>

type Key = keyof typeof defaultJourneys;

type Match<route extends string, B extends string = ''> = `#/${route}/${Journey}` | `?${B}${string}`

type Setup = { journey?: Journey, key: JourneyKeys, match: ReadonlyArray<Match<"service", "journey">>; }

const defaultJourneys = {
  resetPassword: {
    journey: 'ResetPassword',
    key: 'resetPassword',
    match: ['#/service/ResetPassword'] as const,
  },
  forgottenUsername: {
    journey: 'ForgottenUsername',
    key: 'forgottenUsername',
    match: ['#/service/ForgottenUsername'] as const,
  },
  login: {
    journey: undefined,
    key: 'login',
    match: ['#/service/Login', '?journey'] as const,
  },
  registration: {
    journey: 'Registration',
    key: 'registration',
    match: ['#/service/Registration', '?journey=Registration'] as const,
  },
} satisfies  Record<JourneyKeys, Setup>;

type PartialJourneyKeys = keyof z.infer<typeof partialJourneysSchema>
export function initialize(customJourneys: z.infer<typeof partialJourneysSchema>) {
  if (customJourneys) {
    const arr = Object.keys(customJourneys) as PartialJourneyKeys[]
    return readable(
      arr.map((key) => customJourneys[key]),
    );
  } else {
    const arr = Object.keys(defaultJourneys) as Key[];
    return readable(
      arr.map(
        (key) => {
          return defaultJourneys[key];
        },
      ),
    );
  }
}
export let configuredJourneys: Readable<z.infer<typeof journeyItemSchema>[]
```